### PR TITLE
Fix error when receiving status only responses from fetchApi

### PR DESCRIPTION
**Pull request by mpavlinsky**

### CODE REVIEW

File: src/fetchApi.js [hunk 1]
```diff
@@ -14,7 +14,11 @@ export async function fetchApi(request) {
   })
     .then(async (response) => {
       const { headers, status } = response;
-      const responseBody = await response.json();
+      let responseBody;
+      const contentType = response.headers.get('content-type').toLowerCase();
+      if (contentType && contentType.includes('application/json')) {
+        responseBody = await response.json();
+      }
       return { request, status, headers, body: responseBody };
```
Comment (djMax) [inline src/fetchApi.js]: so in the case of non-JSON, the body is just empty... Is that what we want or should we parse it raw?

Comment (mpavlinsky) [reply src/fetchApi.js]: Can we parse it if it's empty?

What should we parse it with, https://developer.mozilla.org/en-US/docs/Web/API/Body/blob ?

Comment (mpavlinsky) [reply src/fetchApi.js]: Oh I understand now.
Will the other parses not throw if the body is empty as well?

Comment (djMax) [reply src/fetchApi.js]: No, they shouldn't throw. And yeah, probably as a blob

**Review by djMax — APPROVED** [review]

**Pull request merged**